kolibri/plugins/user_profile/assets/src/views/ChangeFacility/SelectFacility.vue

```diff
@@ -120,39 +117,74 @@
         return savedAddressesInitiallyFetched.value && discoveredAddressesInitiallyFetched.value;
       });
       const storageFacilityId = useLocalStorage('kolibri-lastSelectedFacilityId', '');
+
+      // data:
+      const availableAddressIds = ref([]);
+      const availableFacilities = ref([]);
+      const selectedFacilityId = ref('');
+      const showAddAddressModal = ref(false);
+
+      // computed properties (functions):
+      const { isMinimumKolibriVersion } = useMinimumKolibriVersion();
+      const facilityDisabled = computed(() => {
+        return function(facility) {
+          return (
+            discoveryFailed.value ||
+            availableAddressIds.value.find(id => id == facility.address_id) === undefined ||
+            !isMinimumKolibriVersion.value(facility.kolibri_version, 0, 16, 0)
+          );
+        };
+      });
+
+      // methods:
+      function createSnackbar(args) {
+        this.$store.dispatch('createSnackbar', args);
+      }
+
+      function handleAddedAddress() {
+        refreshSavedAddressList();
+        createSnackbar(this.$tr('addAddressSnackbarText'));
+      }
+
+      function resetSelectedAddress() {
+        const enabledFacilities = availableFacilities.value
+          .map(f => f.id)
+          .find(f => isMinimumKolibriVersion.value(f.kolibri_version, 0, 16));
```

Comment:
I think that here you're searching through an array of ids, due to the previous `map`, so `f.kolibri_version` will never be defined